pydra/engine/environments.py

```diff
@@ -0,0 +1,84 @@
+from .helpers import execute
+
+from pathlib import Path
+
+
+class Environment:
+    def setup(self):
+        pass
+
+    def execute(self, task):
+        raise NotImplementedError
+
+    def teardown(self):
+        pass
+
+
+class Native(Environment):
+    def execute(self, task):
+        # breakpoint()
+        # args = task.render_arguments_in_root()
+        keys = ["return_code", "stdout", "stderr"]
+        values = execute(task.command_args(), strip=task.strip)
+        output = dict(zip(keys, values))
+        if output["return_code"]:
+            msg = f"Error running '{task.name}' task with {task.command_args()}:"
+            if output["stderr"]:
+                msg += "\n\nstderr:\n" + output["stderr"]
+            if output["stdout"]:
+                msg += "\n\nstdout:\n" + output["stdout"]
+            raise RuntimeError(msg)
+        return output
+
+
+class Docker(Environment):
+    def __init__(
+        self, image, tag="latest", binds=None, output_cpath="/output_pydra", xargs=None
+    ):
+        self.image = image
+        self.tag = tag
+        self.xargs = xargs
+        self.output_cpath = output_cpath
+
+    @staticmethod
+    def bind(loc, mode="ro", root="/mnt/pydra"):  # TODO
+        # XXX Failure mode: {loc} overwrites a critical directory in image
```

Comment:
@effigies - not sure if I understand this comment. Do you remember what you had in mind? is it still relevant?

Reply:
I think this comment is out of date.